src/main/java/seedu/address/model/consultation/UniqueConsultationList.java

```diff
@@ -37,7 +37,7 @@
      */
     public boolean contains(Consultation toCheck) {
         requireNonNull(toCheck);
-        return internalConsults.stream().anyMatch(toCheck::isSameConsultation);
+        return internalConsults.stream().anyMatch(consultation -> consultation.isSameConsultation(toCheck));
```

Comment:
What's the difference between using `consultation` vs using `toCheck` to check for equality? Does the caller matter? Feels like I'd prefer the previous version where we call the method reference directly.

Reply:
I think you're right, but for some reason, when I was trying to disable users from adding personal consults from different students at the same time, this way of referencing seemed to work. Else failing the test cases and actual result.

Reply:
Here's what I'm talking about:

With `internalConsults.stream().anyMatch(toCheck::isSameConsultation)`
<img width="900" alt="Screenshot 2020-11-08 at 11 11 08 AM" src="https://user-images.githubusercontent.com/49579253/98456042-64446a80-21b3-11eb-9866-c9ef28120452.png">

With the **new one**:
<img width="901" alt="Screenshot 2020-11-08 at 11 12 35 AM" src="https://user-images.githubusercontent.com/49579253/98456049-850cc000-21b3-11eb-99e7-43aa37779cfc.png">

Reply:
Hmmm feels odd to me. Feels like equality for Consultations should be transitive (i.e. if A is the same consult as B, B should be the same consult as A). This might be indicative of a larger bug that we should triage, with regards to the equality of Consultations. Let's triage this with another issue & PR.

Otherwise, can you please leave a comment in the code to indicate why you need to write the code like that, perhaps name it "tech debt" somewhere, so that others know why it's written like that, and we can ensure that we come back and pick it up again?

Reply:
Tracked in https://github.com/AY2021S1-CS2103T-F11-1/tp/issues/178